src/Utilities/box.jl

```diff
@@ -39,72 +8,265 @@ Stores the constants of scalar constraints with the lower bound of the set in
 `lower` and the upper bound in `upper`.
 """
 struct Box{T}
+    set_mask::Vector{UInt8}
```

Comment:
It seems weird that MatrixOfConstraints would have this `set_mask` because it is part the info normally set in the `set` field.
However, this `set_mask` is exactly `MixOfScalarSets`. If we were using `MixOfScalarSets` here (that we improve to use bitwise operations), it would allow `GenericModel` to be also generic on the `SingleVariable` constraints.
What would you think of having a VariableConstraints struct containing a Box and a MixOfScalarSets ?

Reply:
What's weird is that we have this `Box` type for MatrixOfConstraint, but we also used it as the variable constraint field in Model.

The abstractions already broken, so what's the harm in the extra `set_mask` field?

Reply:
> The abstractions already broken

Why is it weird or broken ? A `Box` is a hyperrectangle, you can have both a vector of variables or a vector of affine constraints belong to a hyperrectangle, there is nothing weird about that